salt/_modules/caasp_hosts.py

```diff
@@ -155,15 +154,34 @@ def _add_names(hosts, ips, names):
             hosts[ip] = _concat(hosts[ip], names)
 
 
+def _prepend_names(hosts, ips, names):
+    if not isinstance(names, list):
+        names = [names]
+    if not isinstance(ips, list):
+        ips = [ips]
+
+    for ip in ips:
+        __utils__['caasp_log.debug']('hosts: adding %s -> %s', ip, names)
+        if ip not in hosts:
+            hosts[ip] = names
+        else:
+            hosts[ip] = names + hosts[ip]
+
+
 def _add_names_for(hosts, nodes_dict, infra_domain):
     for id, ifaces in nodes_dict.items():
         ip = __salt__['caasp_net.get_primary_ip'](host=id, ifaces=ifaces)
         if ip:
             _add_names(hosts, ip, [id, id + '.' + infra_domain])
 
+
+def _add_nodenames_for(hosts, nodes_dict, infra_domain):
+    for id, ifaces in nodes_dict.items():
+        ip = __salt__['caasp_net.get_primary_ip'](host=id, ifaces=ifaces)
+        if ip:
             nodename = __salt__['caasp_net.get_nodename'](host=id)
             if nodename:
-                _add_names(hosts, ip, [nodename, nodename + '.' + infra_domain])
+                _prepend_names(hosts, ip, [nodename, nodename + '.' + infra_domain])
```

Comment:
`_add_nodenames_for` and `_add_names_for` are basically the same function: they only differ in the way they obtain the IP. I would define only one function:

```Python
def _get_primary_ip(id, ifaces):
    return __salt__['caasp_net.get_primary_ip'](host=id, ifaces=ifaces)

def _get_nodename(id, ifaces):
    return __salt__['caasp_net.get_nodename'](host=id, ifaces=ifaces)

def _add_names_for(hosts, nodes_dict, infra_domain, getter):
    for id, ifaces in nodes_dict.items():
        ip = getter(host=id, ifaces=ifaces)
        if ip:
            _add_names(hosts, ip, [id, id + '.' + infra_domain])
```